### ISSUE TITLE
Drag&drop in media sources is not regarding filetypes

### ISSUE THREAD

**Issue by Jako** [modxcms/revolution#16278]
### Summary
The drag&drop in media sources is not regarding filetypes.

### Step to reproduce/observed behavior
Create a new media sources which is restricted to i.e. pdf files. Drag a jpg file into the restricted media source. It is moved to that media source, but it is not listed in that media source afterwards.

### Expected behavior
It should not be moved into that media source.

### Environment
MODX 3.0.1

**Comment by smg6511** [comment]
Yes, the ultimate problem here is that the property `allowedFileTypes` is mis-named and has never functioned in the way its name implies. More accurately, it should be called `visibleFileTypes`. Further, you might think you could override the system's `upload_files`, `upload_media`, and `upload_images` settings by redefining them in a media source—but that's not the case. Those are the only settings that provide control over uploadable types. I did a lot of work on this when we were trying to solve the static file issues before the release of 3.0, so I'm pretty familiar with the wacky inner workings of the file system and media source handling. Anyway, what you've identified isn't a bug, per se. 

I think some larger-scale reworking needs to be done in this area but, at the very least, it'd be helpful if we could rename the `allowedFileTypes` property as I suggested above.
